internal/http/errors_test.go

```diff
@@ -0,0 +1,96 @@
+package http
+
+import (
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestWriteErrorJSON(t *testing.T) {
+	for _, test := range []struct {
+		name         string
+		writeFn      func() *httptest.ResponseRecorder
+		expectedCode int
+		expectedBody string
+	}{
+		{
+			name: "basic error",
+			writeFn: func() *httptest.ResponseRecorder {
+				w := httptest.NewRecorder()
+				WriteErrorJSON(w, errors.New("basic error"))
+				return w
+			},
+			// should default to this
+			expectedCode: http.StatusInternalServerError,
+			expectedBody: "{\"error\":\"basic error\"}\n",
+		},
+		{
+			name: "http error",
+			writeFn: func() *httptest.ResponseRecorder {
+				w := httptest.NewRecorder()
+				WriteErrorJSON(w,
+					Error(
+						errors.New("unauthorized"),
+						http.StatusUnauthorized,
+					),
+				)
+				return w
+			},
+			expectedCode: http.StatusUnauthorized,
+			expectedBody: "{\"error\":\"unauthorized\"}\n",
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			w := test.writeFn()
+			require.Equal(t, test.expectedCode, w.Result().StatusCode)
+			require.Equal(t, test.expectedBody, w.Body.String())
+		})
+	}
+}
+
+func TestWriteResponseJSON(t *testing.T) {
+	for _, test := range []struct {
+		name         string
+		writeFn      func() *httptest.ResponseRecorder
+		expectedCode int
+		expectedBody string
```

Comment:
Similar to [here](https://github.com/akuity/kargo/pull/4000/files#r2082688622). The test cases are more less noisy / readable if you just have `code int` and `bodyObj any` here. Create the response recorded in the body of the for loop. Pass code and body to `WriteResponseJSON()`, compare the response code to the one in the test case. Un marshal the response body and compare it to the `bodyObj` in the test case.